integration-tests/actions/vrf/common/actions.go

```diff
@@ -216,3 +220,37 @@ func CreateVRFKeyOnVRFNode(vrfNode *VRFNode, l zerolog.Logger) (*client.VRFKey,
 		Msg("VRF Key created on the Node")
 	return vrfKey, pubKeyCompressed, nil
 }
+
+func FundNodesIfNeeded(ctx context.Context, existingEnvConfig *vrf_common_config.ExistingEnvConfig, client blockchain.EVMClient, l zerolog.Logger) error {
+	if *existingEnvConfig.NodeSendingKeyFundingMin > 0 {
+		for _, sendingKey := range existingEnvConfig.NodeSendingKeys {
+			address := common.HexToAddress(sendingKey)
+			sendingKeyBalance, err := client.BalanceAt(ctx, address)
+			if err != nil {
+				return err
+			}
+			fundingAtLeast := conversions.EtherToWei(big.NewFloat(*existingEnvConfig.NodeSendingKeyFundingMin))
+			fundingToSendWei := new(big.Int).Sub(fundingAtLeast, sendingKeyBalance)
+			fundingToSendEth := conversions.WeiToEther(fundingToSendWei)
+			if fundingToSendWei.Cmp(big.NewInt(0)) == 1 {
+				l.Info().
+					Str("Sending Key", sendingKey).
+					Str("Sending Key Current Balance", sendingKeyBalance.String()).
+					Str("Should have at least", fundingAtLeast.String()).
+					Str("Funding Amount in ETH", fundingToSendEth.String()).
+					Msg("Funding Node's Sending Key")
+				err := actions.FundAddress(client, sendingKey, fundingToSendEth)
+				if err != nil {
+					return err
+				}
+			} else {
+				l.Info().
+					Str("Sending Key", sendingKey).
+					Str("Sending Key Current Balance", sendingKeyBalance.String()).
+					Str("Should have at least", fundingAtLeast.String()).
```

Comment:
These 3 lines repeat above 237-239: consider moving them above the if statement.